clients/githubrepo/branches.go

```diff
@@ -190,7 +190,13 @@ func copyAdminSettings(src *branchProtectionRule, dst *clients.BranchProtectionR
 	copyBoolPtr(src.DismissesStaleReviews, &dst.RequiredPullRequestReviews.DismissStaleReviews)
 	if src.RequiresStatusChecks != nil {
 		copyBoolPtr(src.RequiresStatusChecks, &dst.CheckRules.RequiresStatusChecks)
-		copyBoolPtr(src.RequiresStrictStatusChecks, &dst.CheckRules.UpToDateBeforeMerge)
+		// Workaround for GitHub GraphQL bug https://github.com/orgs/community/discussions/59471
+		// The setting RequiresStrictStatusChecks should tell if the branch is required
+		// to be up to date before merge, but it only returns the correct value if
+		// RequiresStatusChecks is true. If RequiresStatusChecks is false, RequiresStrictStatusChecks
+		// is wrongly retrieved as true.
+		upToDateBeforeMerge := *src.RequiresStatusChecks && *src.RequiresStrictStatusChecks
```

Comment:
is pointer `src.RequiresStrictStatusChecks` guaranteed to be non nil?

Reply:
Yes, both fields `RequiresStatusChecks` and `RequiresStrictStatusChecks` according to GitHub GraphQL API documentation cannot be nil. Also, in my local tests, with admin token, the fields didn't come as nil.

https://docs.github.com/en/graphql/reference/objects#branchprotectionrule
<img width="773" alt="image" src="https://github.com/ossf/scorecard/assets/27388955/2cb4ac25-8093-4d34-a5c1-e76afd721253">

Reply:
I think Laurent was talking about `src` being guaranteed to be non-nil, which would be based on logic.

Reply:
Thanks for the clarification, Spencer! I will take a look.

Reply:
Would this better guarantee I'm not doing an invalid access to `src` pointer?

```suggestion
		if src.RequiresStrictStatusChecks != nil {
			dst.CheckRules.UpToDateBeforeMerge = new(bool)
			*dst.CheckRules.UpToDateBeforeMerge = *src.RequiresStatusChecks && *src.RequiresStrictStatusChecks
		}
```

In the code we have a father validation of `src.RequiresStatusChecks != nil` and I'm adding a validation if `src.RequiresStrictStatusChecks != nil` to then access both values. I believe this would cover what's done in `copyBoolPtr` but performing the AND operation.

Reply:
I personally prefer a mix of what you have now and what you suggested, and I think it resolves @laurentsimon 's question.

```go
if src.RequiresStrictStatusChecks != nil {
    upToDateBeforeMerge := *src.RequiresStatusChecks && *src.RequiresStrictStatusChecks
    copyBoolPtr(&upToDateBeforeMerge, &dst.CheckRules.UpToDateBeforeMerge)
}
```